File: solutions/ruby/etl/1/etl.rb
#
# ETL for scrabbles
#
module BookKeeping
  VERSION = 1
end

# module ETL
module ETL
  module_function

  def transform(old)
    old.each_with_object({}) do |(score, letters), retval|
      letters.each { |letter| retval[letter.downcase] = score }
    end
  end
end

File: solutions/swift/etl/2/Sources/Etl/Etl.swift
class ETL {
  static func transform(_ old: [String: [String]]) -> [String: Int] {
    Dictionary(
      uniqueKeysWithValues: old.flatMap { (key, values) in
        values.map { ($0.lowercased(), Int(key) ?? 0) }
      }
    )
  }
}
